tools/yaml-mapper/mapping_datadog_helm_to_datadogagent_crd.yaml

```diff
@@ -1,455 +1,384 @@
-datadog.apiKeyExistingSecret: spec.global.credentials.apiSecret
-datadog.appKey: spec.global.credentials.appKey
-datadog.appKeyExistingSecret: spec.global.credentials.appSecret
-
-datadog.secretAnnotations:
-datadog.secretBackend.command: 
-datadog.secretBackend.arguments:
-datadog.secretBackend.timeout:
-datadog.secretBackend.enableGlobalPermissions:
-datadog.secretBackend.roles:
-
-datadog.securityContext: spec.global.securityContext
-datadog.securityContext.runAsUser: 
-
-datadog.hostVolumeMountPropagation:
-
-datadog.clusterName: spec.global.clusterName
-
-datadog.site: spec.global.site
-datadog.dd_url: spec.endpoint.url
-datadog.logLevel: spec.global.logLevel
-
-datadog.kubeStateMetricsEnabled:
-
-datadog.kubeStateMetricsNetworkPolicy.create:
-
-datadog.kubeStateMetricsCore.enabled: spec.features.kubeStateMetricsCore.enabled
-datadog.kubeStateMetricsCore.rbac.create:
-datadog.kubeStateMetricsCore.ignoreLegacyKSMCheck:
-datadog.kubeStateMetricsCore.collectSecretMetrics:
-datadog.kubeStateMetricsCore.collectConfigMaps:
-datadog.kubeStateMetricsCore.collectVpaMetrics:
-datadog.kubeStateMetricsCore.collectCrdMetrics:
-datadog.kubeStateMetricsCore.collectApiServicesMetrics:
-datadog.kubeStateMetricsCore.useClusterCheckRunners:
-datadog.kubeStateMetricsCore.labelsAsTags:
-datadog.kubeStateMetricsCore.annotationsAsTags:
-
-datadog.clusterChecks.enabled: spec.features.clusterChecks.enabled
-datadog.clusterChecks.shareProcessNamespace:
-
-datadog.nodeLabelsAsTags: spec.global.nodeLabelsAsTags
-datadog.podLabelsAsTags: spec.global.podLabelsAsTags
-datadog.podAnnotationsAsTags: spec.global.podAnnotationsAsTags
-datadog.namespaceLabelsAsTags: spec.global.namespaceLabelsAsTags
-datadog.namespaceAnnotationsAsTags: spec.global.namespaceAnnotationsAsTags
-datadog.kubernetesResourcesLabelsAsTags: spec.global.kubernetesResourcesLabelsAsTags
-datadog.kubernetesResourcesAnnotationsAsTags: spec.global.kubernetesResourcesAnnotationsAsTags
-datadog.originDetectionUnified.enabled: spec.global.global.originDetectionUnified.enabled
-datadog.tags: spec.global.tags
-datadog.checksCardinality:
-
-datadog.kubelet.host: spec.global.kubelet.host
-datadog.kubelet.tlsVerify: spec.global.kubelet.tlsVerify
-datadog.kubelet.hostCAPath: spec.global.kubelet.hostCAPath
-datadog.kubelet.agentCAPath: spec.global.kubelet.agentCAPath
-datadog.kubelet.podLogsPath:
-datadog.kubelet.coreCheckEnabled: 
-
-datadog.expvarPort:
-
-datadog.dogstatsd.port:
-datadog.dogstatsd.originDetection: spec.features.dogstatsd.originDetectionEnabled
-datadog.dogstatsd.tags:
-datadog.dogstatsd.tagCardinality: spec.features.dogstatsd.tagCardinality
-datadog.dogstatsd.useSocketVolume: spec.features.dogstatsd.unixDomainSocketConfig.enabled
-datadog.dogstatsd.socketPath: spec.features.dogstatsd.unixDomainSocketConfig.path
-datadog.dogstatsd.hostSocketPath:
-datadog.dogstatsd.useHostPort: spec.features.dogstatsd.hostPortConfig.enabled
-# datadog.dogstatsd.useHostPID: # deprecated
```

Comment:
TODO: add handling of deprecated values keys

Reply:
Fixed and I decided to leave the deprecated keys uncommented because users might still be using them.